CMakeLists.txt

```diff
@@ -1,19 +1,16 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.5)
 
 PROJECT(uqmi C)
 
 OPTION(BUILD_STATIC OFF)
 
-ADD_DEFINITIONS(-Os -ggdb -Wall -Werror --std=gnu99 -Wmissing-declarations -Wno-enum-conversion)
+ADD_DEFINITIONS(-Os -ggdb -Wall -Werror --std=gnu99 -Wmissing-declarations -Wno-enum-conversion -Wno-dangling-pointer)
```

Comment:
Overall looks like a nice modernization effort, so while at it, perhaps consider  syncing the compiler options with https://github.com/openwrt/libubox/commit/f804578847de49112d414841df928921e62cc216 (https://github.com/openwrt/ubus/commit/90fb16234c2258e31bc5a0e67a8cff3f30395951, https://github.com/openwrt/ucert/commit/e284ed941972e850951cc11f8065dc4126079daa, https://github.com/openwrt/ustream-ssl/commit/3bc05402bfabc711a335262cbb40664b6dfb1240, https://github.com/openwrt/uci/commit/66264ed9ec9e13321798061782b255ecb4dd64c2 and maybe few others)?

Indeed, its quite dated, we're already on GCC 12/13, so there might be already some newer/better best practices for current compiler versions and we should revisit and maybe consider adapting it.

Reply:
Good idea!